RELEASE_NOTES.md

```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+* preserve checkout history so that `git checkout -` works as expected alongside Git Town commands
```

Comment:
I would like to keep breaking changes at the top. Please move this below that. Please also add a link to #65